### ISSUE TITLE
Add a toon shader

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#16171]
## What problem does this solve or what need does it fill?

Toon shading (also called cel shading) is a very common and popular effect for stylized games, discretizing colors to create an effect like that seen in Windwaker.

![Image](https://github.com/user-attachments/assets/69ffb6f0-f494-4323-9a4e-b1ccbd1694d8)

## What solution would you like?

We create and maintain a high quality example teaching users how to do this.

## What alternative(s) have you considered?

We could add a configurable built-in way to do this.

I think that an example is probably more valuable here, as the level of configurability needed is super high. If it turns out to be popular and feasible, we can later change the example into a feature.

**Comment by IceSentry** [comment]
The extended_material example kinda shows a similar effect. I don't know if it's enough to call it cel shading though.

**Comment by alice-i-cecile** [comment]
Yeah, and it's also not discoverable for the intermediate devs who need a bit of hand-holding to figure out how to achieve this effect :)

**Comment by janhohenheim** [comment]
I can gladly donate [bevy_wind_waker_shader](https://github.com/janhohenheim/bevy_wind_waker_shader), which is nearly exactly the algorithm used for the screenshot in the issue description :)

**Comment by blueforesticarus** [comment]
@janhohenheim does that shader support shadows? 
I notice that https://crates.io/crates/bevy_toon_material exists and supports shadow casting but not receiving.
I also notice your crate supports multiple lights, while bevy_toon_material does not

**Comment by janhohenheim** [comment]
Yes, it uses Bevy's regular lighting under the hood, so all shadow-related things work as expected :)
